### PR TITLE
Log Elasticsearch index counts to Datadog

### PR DESCRIPTION
**Pull request by atsmith813**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
This PR renames the `RecordDbTableCountsWorker` to `RecordDataCountsWorker` and adds logic to also record Elasticsearch index sizes to Datadog.

It should be fine to rename the worker in one go without aliasing this since the current task only runs daily. There shouldn't be any hanging jobs as long as we don't deploy this close to the scheduled run time of that task.

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/projects/6#card-34722425

## Added tests?
- [x] yes

## Added to documentation?
- [x] no documentation needed

## Are there any post deployment tasks we need to perform?
Someone with Heroku access will need to remove the `record_db_table_counts` task from the scheduler and replace it with the `record_db_counts` task.

try_again_gif

### CODE REVIEW

File: app/workers/metrics/record_data_counts_worker.rb [hunk 1]
```diff
@@ -9,6 +9,11 @@ def perform
         estimate = model.estimated_count
         Rails.logger.info("db_table_size", table_info: { table_name: model.table_name, table_size: estimate })
         DatadogStatsClient.gauge("postgres.db_table_size", estimate, tags: { table_name: model.table_name })
+
+        next unless model.const_defined?(:SEARCH_CLASS)
+
+        index_size = Search::Client.count(index: model::SEARCH_CLASS::INDEX_ALIAS).dig("count")
+        DatadogStatsClient.gauge("elasticsearch.index_size", index_size, tags: { table_name: model.table_name })
```
Comment (mstruve) [inline app/workers/metrics/record_data_counts_worker.rb]: Can we rename this gauge `"elasticsearch.document_count"`? Index size to me means the amount of space in bytes the index itself takes up.

File: app/workers/metrics/record_data_counts_worker.rb [hunk 2]
```diff
@@ -9,6 +9,11 @@ def perform
         estimate = model.estimated_count
         Rails.logger.info("db_table_size", table_info: { table_name: model.table_name, table_size: estimate })
         DatadogStatsClient.gauge("postgres.db_table_size", estimate, tags: { table_name: model.table_name })
+
+        next unless model.const_defined?(:SEARCH_CLASS)
+
+        index_size = Search::Client.count(index: model::SEARCH_CLASS::INDEX_ALIAS).dig("count")
```
Comment (mstruve) [inline app/workers/metrics/record_data_counts_worker.rb]: Should we pull this into our `base.rb` search class and then use the model's SEARCH_CLASS to call it? Then we can customize it for FeedContent to allow us to filter those by model type in that index. 
```ruby 
model::SEARCH_CLASS.document_count

# In search/base.rb
def document_count 
  Search::Client.count(index: self.class::INDEX_ALIAS).dig("count")
end
```

Comment (atsmith813) [reply app/workers/metrics/record_data_counts_worker.rb]: Updated. `FeedContent` isn't a model so it's not in the loop, but I'm sure we'll figure that out when the time comes.

**Comment by atsmith813** [comment]
@mstruve do you want to go ahead and deploy this whenever you're ready so you can handle the Heroku setup?

**Comment by rhymes** [comment]
@atsmith813 this new plan is much simpler :D "Log the counts and set an alert", right?

**Pull request merged**